### DISCUSSION TITLE
Incorrect results returned when accessing results of snowflake scripting stored procedure

### DISCUSSION THREAD

**Discussion by bobf32** [dbeaver/dbeaver#15790] [Q&A]
#### System information: 
- Windows 10 Enterprise 21H2 build 19044.1526
- DBeaver version 22.0.0
- Additional extensions: None

#### Connection specification:
- Snowflake 
- Driver name

#### Describe the problem you're observing:

Unable to access return value from stored procedure created with Snowflake scripting 

#### Steps to reproduce, if exist:

create or replace procedure test_sp()
returns table(a float)
language sql
as
$$
    declare
        res RESULTSET default (select 3.141::float a);
    begin
        return table(res);
    end;
$$;

CALL test_sp();

SELECT * FROM table(result_scan(last_query_id()));

#### Include any warning/errors/backtraces from the logs

Expected result: The final SELECT above results in a single field data set being returned, consisting of one field, A, value 3.141. This occurs if running the above code in the Snowflake web front end.

Actual result: In DBeaver, execution results in a two field data set consisting of the current_database() and current_schema() values.

**Comment by LonwoLonwo** [comment]
Hello @bobf32

You can disable this behavior with this setting:
Preferences -> Editors -> SQL Editor -> SQL Processing -> "Refresh active schema after SQL execution"

![2022-03-10 13_04_24-DBeaver Ultimate 22 0 0 - _SNOWFLAKE_SAMPLE_DATA_ Script-28](https://user-images.githubusercontent.com/45152336/157639129-556d2a25-7107-4d3d-a28f-30313499415f.png)

**Comment by AndrewGabelliTR** [comment]
@LonwoLonwo Works perfectly, thank you.
